var/spack/repos/builtin/packages/python/package.py

```diff
@@ -806,6 +734,56 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         pythonpath = ':'.join(python_paths)
         env.set('PYTHONPATH', pythonpath)
 
+        # We want to make sure that the extensions are compiled and linked with
```

Comment:
I wonder whether we should do the following only if `dependent_spec.package.extends(self.spec)`.